sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeRowWriter.java

```diff
@@ -51,6 +51,7 @@ public UnsafeRowWriter(BufferHolder holder, int numFields) {
     this.nullBitsSize = UnsafeRow.calculateBitSetWidthInBytes(numFields);
     this.fixedSize = nullBitsSize + 8 * numFields;
     this.startingOffset = holder.cursor;
+    holder.reset();
```

Comment:
I think we don't guarantee all the calls to this `UnsafeRowWriter` constructor are at the timing of new incoming record. It could be possible that we pass a `BufferHolder` into this constructor but the holder is already written with some data and we want to continue writing from current cursor.